From 9f0e9b81b6224fbfb4813a2303450741505c1886 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Tue, 15 Oct 2024 20:27:24 +0000
Subject: [PATCH] Handle no RootWindow in ClearAuraTransientParent

Handle a crash in gtk::ClearAuraTransientParent when the aura::Window has no host, e.g. the root window is null. This can happen for example when the root window is closed before a print dialog is closed. There is a similar condition in SetGtkTransientForAura, and this allows Electron to remove a patch addressing this crash.

Bug: N/A
Change-Id: Iae0a1ba681cf85360a7630d164f318f0054c828a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5933946
Reviewed-by: Thomas Anderson <thomasanderson@chromium.org>
Reviewed-by: Nick Yamane <nickdiego@igalia.com>
Commit-Queue: Nick Yamane <nickdiego@igalia.com>
Cr-Commit-Position: refs/heads/main@{#1368999}
---

--- a/ui/gtk/gtk_util.cc
+++ b/ui/gtk/gtk_util.cc
@@ -227,9 +227,13 @@
 }
 
 void ClearAuraTransientParent(GtkWidget* dialog, aura::Window* parent) {
+  if (!parent || !parent->GetHost()) {
+    return;
+  }
+
   g_object_set_data(G_OBJECT(dialog), kAuraTransientParent, nullptr);
-  GtkUi::GetPlatform()->ClearTransientFor(
-      parent->GetHost()->GetAcceleratedWidget());
+  gfx::AcceleratedWidget parent_id = parent->GetHost()->GetAcceleratedWidget();
+  GtkUi::GetPlatform()->ClearTransientFor(parent_id);
 }
 
 base::OnceClosure DisableHostInputHandling(GtkWidget* dialog,
